examples/library/test_manycore_dmem_read_write.cpp

```diff
@@ -56,6 +56,7 @@ int test_mem_functions(hb_mc_manycore_t *mc) {
         }
         bsg_pr_test_info("Write successful\n");
 
+        // TODO: We need to move available IDs to global.
```

Comment:
Did I write this?

Reply:
yes

Reply:
OK, less of a comment for this test, and more of a comment for the library. TODO: Remove this comment and file an issue to track outstanding IDs at the pod (?) or device (?) level

Reply:
I do this in the network simulator with a set, but there are many ways to do this.

Reply:
I think we do it now with an std::map (this PR) but it's lifetime is at the function scope. CUDA-lite does not support a non-blocking read api. If that were something we wanted to support, then persistent tracking of load ids would be necessary. OTOH that API functionality comes with additional headaches. For one, non-blocking APIs are much harder to use.

Reply:
Yeah - I think I'm more concerned about a user manually sending packets, and then using the read/write API. 

I still feel that we should file an issue and track it. Would be an easy fix.